code/modules/mob/living/silicon/robot/emote.dm

```diff
@@ -207,17 +224,31 @@
 			else
 				to_chat(src, "You are not security.")
 
-		/*
-		if ("fart")
-			var/list/robotfarts = list("makes a farting noise","vents excess methane","shakes violently, then vents methane.")
-			var/robofart = pick(robotfarts)
-			message = "<B>[src]</B> [robofart]."
-			m_type = VISIBLE
-
-		*/
+		if ("vent")
+			if(world.time-lastvent >= 30 SECONDS)
+				var/list/robotfarts = list("vents","shakes violently, then vents")
+				var/robofart = pick(robotfarts)
+				for(var/mob/living/M in loc)
+					if(M != src && M.loc == src.loc && (M.lying || M.resting))
```

Comment:
Since you're looping through Loc now, checking whether the Loc is the same as src is redundant.